extension/ackextension/inmemory.go

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ackextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/ackextension"
+
+import (
+	"context"
+	"sync/atomic"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+	"go.opentelemetry.io/collector/component"
+)
+
+// InMemoryAckExtension is the in-memory implementation of the AckExtension
+// When MaxNumPartition is reached, the acks associated with the least recently used partition are evicted.
+// When MaxNumPendingAcksPerPartition is reached, the least recently used ack is evicted
+type InMemoryAckExtension struct {
```

Comment:
Please don't export it, there is no need

Reply:
fixed